tensorflow/compiler/mlir/tosa/transforms/convert_tfl_uint8.cc

```diff
@@ -130,6 +130,63 @@ struct ConvertUint8QConstOp : public RewritePattern {
   }
 };
 
+namespace {
+
+// returns true iff @a type is a shaped type with element type that is uint8
+// if it is, then return the rescaled type, uint8_zp, and output_zp to use to
+// rescale type to signed type with adjusted zero point.
+bool IsShapedUint8Type(OpBuilder& builder, const Type type, Type& rescaled_type,
```

Comment:
`Type` should be `ShapedType`. I would expect your lowering to require `ShapedType` for all of its lowering and would fail otherwise. Better to restrict the helper to only use expected cases.

The name is also bad, it should indicate that it is intended to extract type information. `IsShapedUint8Type` makes it seem like it only returns `true` or `false` when it actually extracts additional information.

Reply:
fixed